### DISCUSSION TITLE
WhatsApp OTP Support

### DISCUSSION THREAD

**Discussion by elenamik** [supabase/supabase#12586] [Feature Requests]
We’re having some issues with SMS delivery in emerging markets and want to give users an option to get WhatsApp messages instead (Twillio [supports](https://www.twilio.com/blog/whatsapp-verification-code-5-minutes) that).

Working to implement manually with [roll-your-own-auth](https://github.com/burggraf/supabase-roll-your-own-auth), but would be helpful if it was hooked up to the auth API

[edit] As a note - the most user friendly method for us would be a magic link sent to the user's WhatsApp to avoid the user having to copy and paste a code themselves

**Comment by ioucyf** [comment]
+1 🙏

**Comment by elenamik** [comment]
being worked on [here](https://github.com/supabase/gotrue-js/pull/616) :)

**Comment by kangmingtay** [comment]
resolving this one because this is now support - https://supabase.com/docs/guides/auth/phone-login/twilio#whatsapp-otp-logins
